openadapt/db.py

```diff
@@ -51,3 +55,154 @@ def get_base(engine):
+
+    Returns:
+        sql (str): The SQL statement to insert the recording into the output file.
+    """
+    engine = sa.create_engine(config.DB_URL)
+    Session = sessionmaker(bind=engine)
+    session = Session()
+
+    recording = get_recording_by_id(recording_id)
+
+    if recording:
+        sql = """
+            INSERT INTO recording
+            VALUES (?, ?, ?, ?, ?, ?, ?, ?)
+        """
+        values = (
+            recording.id,
+            recording.timestamp,
+            recording.monitor_width,
+            recording.monitor_height,
+            recording.double_click_interval_seconds,
+            recording.double_click_distance_pixels,
+            recording.platform,
+            recording.task_description,
+        )
+
+        logger.info(f"Recording with ID {recording_id} exported successfully.")
+    else:
+        sql = ""
+        logger.info(f"No recording found with ID {recording_id}.")
+
+    return sql, values
+
+
+def update_db_fname_in_env_file(db_fname: str) -> None:
+    """
+    Update the database filename in the environment file.
+
+    Args:
+        db_fname (str): The new filename for the database.
+
+    Returns:
+        None
+    """
+    with open(config.ENV_FILE_PATH, "r") as env_file:
+        env_file_lines = [
+            f"DB_FNAME={db_fname}\n"
+            if env_file_line.startswith("DB_FNAME")
+            else env_file_line
+            for env_file_line in env_file.readlines()
+        ]
+
+    with open(config.ENV_FILE_PATH, "w") as env_file:
+        env_file.writelines(env_file_lines)
+
+
+def create_db(recording_id: int, sql: str, values) -> tuple[float, str]:
+    """Create a new database and import the recording data.
+
+    Args:
+        recording_id (int): The ID of the recording.
+        sql (str): The SQL statements to import the recording.
+
+    Returns:
+        tuple: A tuple containing the timestamp and the file path of the new database.
+    """
+    db_file_path = None
+    target_file_path = None
+    try:
+        db_fname = f"recording_{recording_id}.db"
+        original_db_fname = config.DB_FNAME  # Store the original value
+
+        timestamp = time.time()
+        source_file_path = config.ENV_FILE_PATH
+        target_file_path = f"{config.ENV_FILE_PATH}-{timestamp}"
+        logger.info(
+            f"source_file_path={source_file_path}, target_file_path={target_file_path}"
+        )
+        shutil.copyfile(source_file_path, target_file_path)
+        config.set_db_url(db_fname)
+        update_db_fname_in_env_file(db_fname)
+        db_file_path = config.DB_FPATH.resolve()
+
+        engine = sa.create_engine(config.DB_URL)
+        Session = sessionmaker(bind=engine)
+        session = Session()
+        os.system("alembic upgrade head")
+
+        with engine.begin() as connection:
+            connection.execute(sql, values)
+
+        return timestamp, db_file_path
+
+    except KeyboardInterrupt:
```

Comment:
except Exception

Reply:
addressed in https://github.com/OpenAdaptAI/OpenAdapt/pull/223/commits/3992d1c78d334a352054c2575db248df8677ec4a